index_release_test.go

```diff
@@ -17,6 +22,1049 @@ func urlMustParse(v string) *URL {
+					},
+					timestamp: "2018-05-22T12:00:00.000000Z",
+					version:   "3.0.0",
+				},
+			},
+		},
+		{
+			name: "case 2: introduction of patch to bar-operator",
+			releases: []Release{
+				{
+					changelogs: []Changelog{
+						{
+							Component:   "foo-operator",
+							Description: "new feature x",
+							Kind:        KindAdded,
+						},
+						{
+							Component:   "bar-operator",
+							Description: "new feature y",
+							Kind:        KindAdded,
+						},
+					},
+					timestamp: "2018-05-02T12:00:00.000000Z",
+					version:   "1.0.0",
+				},
+				{
+					changelogs: []Changelog{
+						{
+							Component:   "foo-operator",
+							Description: "new feature x",
+							Kind:        KindAdded,
+						},
+						{
+							Component:   "bar-operator",
+							Description: "changed feature y",
+							Kind:        KindChanged,
+						},
+					},
+					timestamp: "2018-05-28T12:00:00.000000Z",
+					version:   "1.0.1",
+				},
+				{
+					changelogs: []Changelog{
+						{
+							Component:   "foo-operator",
+							Description: "new feature x",
+							Kind:        KindAdded,
+						},
+						{
+							Component:   "bar-operator",
+							Description: "new feature y",
+							Kind:        KindAdded,
+						},
+						{
+							Component:   "baz-operator",
+							Description: "new feature quux",
+							Kind:        KindAdded,
+						},
+					},
+					timestamp: "2018-05-22T12:00:00.000000Z",
+					version:   "2.0.0",
+				},
+				{
+					changelogs: []Changelog{
+						{
+							Component:   "foo-operator",
+							Description: "new feature x",
+							Kind:        KindAdded,
+						},
+						{
+							Component:   "bar-operator",
+							Description: "changed feature y",
+							Kind:        KindChanged,
+						},
+						{
+							Component:   "baz-operator",
+							Description: "new feature quux",
+							Kind:        KindAdded,
+						},
+					},
+					timestamp: "2018-05-28T15:00:00.000000Z",
+					version:   "2.0.1",
+				},
+			},
+			expectedReleases: []Release{
+				{
+					changelogs: []Changelog{
+						{
+							Component:   "foo-operator",
+							Description: "new feature x",
+							Kind:        KindAdded,
+						},
+						{
+							Component:   "bar-operator",
+							Description: "new feature y",
+							Kind:        KindAdded,
+						},
+					},
+					version: "1.0.0",
+				},
+				{
+					changelogs: []Changelog{
+						{
+							Component:   "bar-operator",
+							Description: "changed feature y",
+							Kind:        KindChanged,
+						},
+					},
+					version: "1.0.1",
+				},
+				{
+					changelogs: []Changelog{
+						{
+							Component:   "baz-operator",
+							Description: "new feature quux",
+							Kind:        KindAdded,
+						},
+					},
+					version: "2.0.0",
+				},
+				{
+					changelogs: []Changelog{
+						{
+							Component:   "bar-operator",
+							Description: "changed feature y",
+							Kind:        KindChanged,
+						},
+					},
+					version: "2.0.1",
+				},
+			},
+		},
```

Comment:
I thougth of a case where I am not sure how we handle this or if we allow this to happen:
 - version `1.0.0` is released
 -  version `1.1.0` is released
 - now patch `1.0.1` is released

Can this happen in this timeline? Or would we always force the patch to be `1.1.1` ?

Reply:
This is what the last test case is about. At least I understood it that way. Do you mean the same or where is the difference?

Reply:
Oh you are right, I misread the last testcase and didn't look at the timestamp closely enough.

Ignore this comment.